src/jobflow_remote/jobs/jobcontroller.py

```diff
@@ -4951,30 +4951,93 @@ def _cancel_queue_process(self, job_doc: dict) -> None:
 
     @staticmethod
     def _build_query_batch(
+        process_id: str | list[str] | None = None,
+        batch_uid: str | list[str] | None = None,
         worker: str | list[str] | None = None,
         batch_state: BatchState | list[BatchState] | None = None,
```

Comment:
Maybe interesting (here but then also to be added in the cli of course) to also have the possibility to get the list (should be only 1 in principle) that contains a given job_id ? and/or job_id+index ? and/or db_id ? (for the latter case, I think it would not be possible right now without an aggregation, except if we add the db_id to the info field of the BatchDoc). What do you think ?

Reply:
For querying on jobid+index, isn't that just something like "info.SOMEID.SOMEINDEX": exists: True ? (I'm on my phone sorry for the lack of correctness 😃)

For the db id, probably good to add indeed. Maybe in the info dict? (But the query is not so easy then) or just as a list at root level ? (But then the correspondance with jobid+index is not known from the inside of the BatchDox).

Info should indeed be typed as a dict.